examples_new/common/player.cpp

```diff
@@ -49,10 +49,14 @@ void Player::sendToServer(std::string message, std::string arguments) {
 }
 
 json Player::receive() {
-  char buffer[4096];
-  memset(buffer, '\0', sizeof(buffer));
-  int ret = read(mConnFd, (void *)buffer, sizeof(buffer) - 1);
-  return json::parse(buffer);
+  std::string completeBuffer;
+  do {
+    char buffer[4096];
```

Comment:
I would set a buffer size of 4097 (so that we actually read 4096 bytes from the socket) and still have the final `\0`.